src/components/Avatar/AvatarStack/AvatarStack.style.ts

```diff
@@ -0,0 +1,27 @@
+import { css, SerializedStyles } from '@emotion/react';
+import { rem } from 'theme/utils';
+
+import { AvatarSizes } from '../Avatar';
+
+const iconMarginLeft = (size: AvatarSizes) => {
+  switch (size) {
+    case 'lg':
+      return rem(-8);
```

Comment:
🤔

Reply:
Why are negative values used here? Can we find a cleaner workaround here?

Reply:
@Christos-Zacharopoulos They are overlapping here, so negative margins are one way to do this. Would you suggest doing it another way?

Reply:
The overlap is handled by the z-index. The problem is the positioning of the items. Right now they are placed outside of their parent container. Also,

1) the first one is outside the container due to the negative margin. 
![image](https://user-images.githubusercontent.com/28539607/148770301-fe545a8c-502f-4857-898d-8286fe93df10.png)
This is easily fixed.

2) This has the disadvantage of having the margin of each size of them hard-copied. This is easily forgotten and might create issues in the future. 

3) Using negative values is like trying to find a dirty workaround for a problem. Each element can use it's dedicated space without having to do this. I prefer to use this kind of solutions for styling third party libraries that I absolutely need and can't find an alternative.

Proposed way:

This might need a few more changes but it's cleaner and helps safeguard future changes.

We can define the width instead of the margin-left in `avatarStackStyle` of all the containers except for the last one. This way everything is nice and tidy inside their container. 

I would also like to see more consistency in the amount of overlap each items has.

For example: 
`lg` size is 46 but overlap is 8.
`md` size is 24 but overlap is 4.
`sm` size is 20 but overlap is 4.
`xs` size is 16 but overlap is 2.

There is no rule or consistency in the amount of overlap. We could propose to the design team for the overlap to be 1/3 or 1/4 of the avatar's size. And it's also very easy.

In `Avatar.styles.ts`  there is a `sizeBasedOnProp`  function. 

If you refactor this to 
`const sizeBasedOnProp = (size: AvatarSizes) => {
  switch (size) {
    case 'md':
      return 24;
    case 'sm':
      return 20;
    case 'xs':
      return 16;
    default:
      return 46;
  }
};`

then you can use this to also set the width of the `avatarStackStyle`